test/account/AccountExecHooks.t.sol

```diff
@@ -50,13 +36,7 @@ contract AccountExecHooksTest is OptimizedTest {
     event ReceivedCall(bytes msgData, uint256 msgValue);
 
     function setUp() public {
-        entryPoint = new EntryPoint();
```

Comment:
Hmm I think this changes things up, setUp() is called before each test whereas the constructor only runs once. I think with these changes, if a test installs a plugin, the installed plugin would persist in the next test. We've not seen any reverts because we deploy a new MockPlugin contract instance per test and set that as a global MockPlugin

Reply:
If the above is right, tests are no longer independent, and reasoning the correctness of a certain functionality is harder

Reply:
They're all run in forked environments, and the forking point happens after calling `setUp`, so it doesn't make a difference. Just to confirm, I ran this test and it passed.

```solidity
contract BaseContract is Test {
    Counter counter1;

    constructor() {
        counter1 = new Counter();
    }

}

contract FoundryTestBehaviorTest is BaseContract {
    
    function setUp() public {
        counter1.increment();
    }

    function test_behav_1() public {
        assertEq(counter1.number(), 1);
        counter1.increment();
        assertEq(counter1.number(), 2);
    }

    function test_behav_2() public {
        assertEq(counter1.number(), 1);
        counter1.increment();
        assertEq(counter1.number(), 2);
    }

}
```